### PR TITLE
Refactor audioSegmentation and audioTrainTest

### PR DESCRIPTION
**Pull request by dsgou**
I have not completed the test script because i did not have the data, run it and let me know if something goes wrong.

### CODE REVIEW

File: pyAudioAnalysis/audioAnalysis.py [hunk 1]
```diff
@@ -201,36 +201,36 @@ def segmentclassifyFileWrapper(inputWavFile, model_name, model_type):
...
         gtFile = inputWavFile.replace(".mp3", ".segments")
-    aS.mtFileClassification(inputWavFile, model_name, model_type, True, gtFile)
+    aS.mid_term_file_classification(inputWavFile, model_name, model_type, True, gtFile)
 
 
 def segmentclassifyFileWrapperHMM(wavFile, hmmModelName):
     gtFile = wavFile.replace(".wav", ".segments")
-    aS.hmmSegmentation(wavFile, hmmModelName, plot_res=True,
-                       gt_file_name=gtFile)
+    aS.hmm_segmentation(wavFile, hmmModelName, plot_res=True,
```
Comment (tyiannak) [inline pyAudioAnalysis/audioAnalysis.py]: This throws an error (the correct name here is plot_results)

**Comment by tyiannak** [comment]
There are still some bugs. 
1) segment and classify file fails (probably at evaluation). To reproduce the error run: 
python3 audioAnalysis.py segmentClassifyFile -i data/scottish.wav --model knn --modelName sm_knn 
2) same for HMM tests

**Comment by dsgou** [comment]
Tests 12 should be ok now, i have not yet run number 6

**Comment by tyiannak** [comment]
thanx 6 and 12 seem to pass now, i'll run the shell tests and python tests one more and merge tomorrow if everything is ok

**Pull request merged**